src/main/java/com/networknt/schema/EnumValidator.java

```diff
@@ -1,75 +1,88 @@
-
-        parseErrorCode(getValidatorType().getErrorCodeKey());
-    }
-
-    public Set<ValidationMessage> validate(JsonNode node, JsonNode rootNode, String at) {
-        debug(logger, node, rootNode, at);
-
-        Set<ValidationMessage> errors = new LinkedHashSet<ValidationMessage>();
-
-        if (!nodes.contains(node)) {
-            errors.add(buildValidationMessage(at, error));
-        }
-
-        return Collections.unmodifiableSet(errors);
-    }
-
-}
+/*
+ * Copyright (c) 2016 Network New Technologies Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.networknt.schema;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.HashSet;
+import java.util.Set;
+
+public class EnumValidator extends BaseJsonValidator implements JsonValidator {
+    private static final Logger logger = LoggerFactory.getLogger(EnumValidator.class);
+
+    private final Set<JsonNode> nodes;
+    private final String error;
+
+    public EnumValidator(String schemaPath, JsonNode schemaNode, JsonSchema parentSchema, ValidationContext validationContext) {
+            super(schemaPath, schemaNode, parentSchema, ValidatorTypeCode.ENUM, validationContext);
+
+        if (schemaNode != null && schemaNode.isArray()) {
+            nodes = new HashSet<JsonNode>();
+            StringBuilder sb = new StringBuilder();
+
+            sb.append('[');
+            String separator = "";
+
+            for (JsonNode n : schemaNode) {
+                nodes.add(n);
+
+                sb.append(separator);
+                sb.append(n.asText());
+                separator = ", ";
+            }
+
+            sb.append(']');
+
+            error = sb.toString();
+        } else {
+            nodes = Collections.emptySet();
+            error = "[none]";
+        }
+
+        parseErrorCode(getValidatorType().getErrorCodeKey());
+    }
+
+    public Set<ValidationMessage> validate(JsonNode node, JsonNode rootNode, String at) {
+        debug(logger, node, rootNode, at);
+
+        Set<ValidationMessage> errors = new LinkedHashSet<ValidationMessage>();
+
+        if (!nodes.contains(node) && !isTypeLooseEqual(node)) {
+            errors.add(buildValidationMessage(at, error));
+        }
+
+        return Collections.unmodifiableSet(errors);
+    }
+
+    private boolean isTypeLooseEqual(JsonNode node) {
```

Comment:
1. I think it's doing a lot more then the name would suggest; It's checking if the type is loose, checking if the parameter is a string node, and checking if that string value appears in the nodes instance variable (the enum).. similar to an "contains" check. Maybe we can rename it to something like "isTypeLooseAndInEnum"? or move the typeLoose check out of it and just have this method check if the node is a string and in the enum?
2. Would it make sense to some unit tests to it?

Reply:
Yes, this name does not quite match its function. I will modify it. I feel that taken out typeLoose check separately to be more understandable and more consistent with the judgments in other verifications.

I will try to add some testcase about enum
For example, enter a number, enter a string, and include a number or string in the enum. Test if it works in the case of typeLoose.